### PR TITLE
Add VLAN test with tagged ports

### PR DESCRIPTION
**Pull request by StepanVovkPLV**
Tests added 
```
test_tagged_broadcast
test_tagged_multicast
test_tagged_unicast
```
NOTE: Baased on [PR](https://github.com/PLVision/dent-testing/pull/16)
Signed-off-by: Stepan Vovk <stepan.vovk@plvision.eu>

### CODE REVIEW

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_tagged_ports.py [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+            continue
+        for setting in config["settings"]:
+            streams_to_learn.update({f"{tg_ports[config['port']]} -> setting VLAN {setting['vlan']}": {
+                "type": "raw",
+                "protocol": "802.1Q",
+                "vlanID": setting["vlan"],
+                "ip_source": dev_groups[tg_ports[config['port']]][0]["name"],
+                "ip_destination": dev_groups[tg_ports[tx_port]][0]["name"],
+                "srcMac": mac_addr[tg_ports[config["port"]]],
+                "dstMac": tx_port_mac
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_tagged_ports.py]:
```suggestion
                "dstMac": tx_port_mac,
                "rate": 10,  # pps
```

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_tagged_ports.py [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+            }})
+    #  Untagged stream
+    streams.update({f"{tg_ports[0]} -> untagged": {
+        "type": "raw",
+        "protocol": "802.1Q",
+        "ip_source": tx_ports,
+        "ip_destination": rx_ports,
+        "srcMac": tx_port_mac,
+        "dstMac": mac_addr[tg_ports[3]]
+    }})
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_tagged_ports.py]:
```suggestion
    for rx_port, mac in mac_addr.items():
        if mac == tx_port_mac:
            continue
        for vlan in packet_vids:
            streams.update({f"{tg_ports[0]} -> {rx_port}, VLAN: {vlan}": {
                "type": "raw",
                "protocol": "802.1Q",
                "ip_source": tx_ports,
                "ip_destination": rx_ports,
                "srcMac": tx_port_mac,
                "dstMac": mac,
                "vlanID": vlan,
            }})
        #  Untagged stream
        streams.update({f"{tg_ports[0]} -> {rx_port}, untagged": {
            "type": "raw",
            "protocol": "802.1Q",
            "ip_source": tx_ports,
            "ip_destination": rx_ports,
            "srcMac": tx_port_mac,
            "dstMac": mac,
        }})
```

**Review by SerhiyBoikoPLV — CHANGES_REQUESTED** [review]

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/constants.py [hunk 3]
```diff
@@ -104,5 +105,8 @@
     "suite_group_connection": ["suite_connection"],
     "suite_group_platform": ["suite_poe", "suite_onlpdump", "suite_lldp"],
     "suite_group_cleanup": ["suite_cleanup"],
-    "suite_group_functional": ["suite_functional_ipv4"],
+    "suite_group_functional": [
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/constants.py]: there is one more `suite_group_functional` at Line 77 that you should remove

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_all_supported_modes.py [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+                                                         tgen_utils_stop_traffic, tgen_utils_get_traffic_stats,
+                                                         tgen_utils_get_loss, tgen_utils_dev_groups_from_config,
+                                                         tgen_utils_stop_protocols,
+                                                         tgen_utils_traffic_generator_connect, )
+from dent_os_testbed.utils.test_utils.br_utils import (configure_bridge_setup, configure_vlan_setup,
+                                                       get_traffic_port_vlan_mapping)
+
+pytestmark = [pytest.mark.suite_functional_vlan,
+              pytest.mark.asyncio,
+              pytest.mark.usefixtures("cleanup_bridges")]
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_all_supported_modes.py]:
```suggestion
              pytest.mark.usefixtures("cleanup_bridges", "cleanup_tgen")]
```

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_all_supported_modes.py [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+    stats = await tgen_utils_get_traffic_stats(tgen_dev, "Flow Statistics")
+    ti_to_rx_port_map = get_traffic_port_vlan_mapping(streams, port_map, tg_ports)
+    for row in stats.Rows:
+        if row["Rx Port"] in ti_to_rx_port_map[row["Traffic Item"]]:
+            assert tgen_utils_get_loss(row) == 0.000, \
+                f'Traffic leak for traffic item: {row["Traffic Item"]} on port {row["Rx Port"]}'
+        else:
+            assert tgen_utils_get_loss(row) == 100.000, \
+                f'No traffic for traffic item : {row["Traffic Item"]} on port {row["Rx Port"]}'
+    await tgen_utils_stop_protocols(tgen_dev)
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_all_supported_modes.py]: use `cleanup_tgen`

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_all_supported_modes.py [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+            "srcMac": vlan_to_mac[port['settings'][0]['vlan']],
+            "dstMac": mac[tg_ports[0]],
+            "vlanID": port['settings'][0]['vlan']
+        }})
+
+    streams.update(streams_to_learn)
+    await tgen_utils_setup_streams(tgen_dev, config_file_name=None, streams=streams)
+
+    # 5. Send traffic to rx_ports.
+    await tgen_utils_start_traffic(tgen_dev)
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_all_supported_modes.py]:
```suggestion
    # 5. Send traffic to rx_ports.
    # learning
    await tgen_utils_start_traffic(tgen_dev)
```

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_tagged_ports.py [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+                                     {"vlan": 24, "untagged": False, "pvid": True}]},
+
+            {"port": 2, "settings": [{"vlan": 22, "untagged": False, "pvid": False},
+                                     {"vlan": 23, "untagged": False, "pvid": False},
+                                     {"vlan": 24, "untagged": False, "pvid": True}]},
+
+            {"port": 3, "settings": [{"vlan": 22, "untagged": False, "pvid": False},
+                                     {"vlan": 23, "untagged": False, "pvid": False},
+                                     {"vlan": 24, "untagged": False, "pvid": True}]
+             })
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_tagged_ports.py]:
```suggestion
                                     {"vlan": 24, "untagged": False, "pvid": True}]},
)
```

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_tagged_ports.py [hunk 8]
```diff
@@ -0,0 +1,292 @@
...
+                "ip_destination": dev_groups[tg_ports[dst]][0]["name"],
+                "srcMac": mac[src],
+                "dstMac": mac[dst],
+                "type": "raw",
+                "vlan": vlan,
+                "protocol": "802.1Q"
+                } for src, dst in ((0, 1), (0, 2), (0, 3))
+            })
+        for port in port_map:
+            if port["port"] == tx_port or isinstance(port["port"], int) is False:
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_tagged_ports.py]: Why do you need `isinstance(port["port"], int)`? When can this be False?

**Review by SerhiyBoikoPLV — CHANGES_REQUESTED** [review]

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_tagged_ports.py (hunk 8 above)
Comment (StepanVovkPLV) [reply DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_tagged_ports.py]: sorry, had to be `isinstance(vlan, int)`

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_tagged_ports.py [hunk 9]
```diff
@@ -0,0 +1,291 @@
...
+            {"port": 2, "settings": [{"vlan": 22, "untagged": False, "pvid": False},
+                                     {"vlan": 23, "untagged": False, "pvid": False},
+                                     {"vlan": 24, "untagged": False, "pvid": True}]},
+
+            {"port": 3, "settings": [{"vlan": 22, "untagged": False, "pvid": False},
+                                     {"vlan": 23, "untagged": False, "pvid": False},
+                                     {"vlan": 24, "untagged": False, "pvid": True}]})
+
+
+async def test_vlan_tagged_ports_with_broadcast(testbed):
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_tagged_ports.py]: How different are those tests? Is it possible to parametrize them?

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_tagged_ports.py [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+            "ip_source": tx_ports,
+            "ip_destination": rx_ports,
+            "srcMac": mac[0],
+            "dstMac": "ff:ff:ff:ff:ff:ff"
+        }})
+    elif traffic_type == "multicast":
+        streams = {f"Traffic with VLAN ID: {vlan}": {
+            "type": "raw",
+            "protocol": "802.1Q",
+            "rate": 10,
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_tagged_ports.py:101]: Why do you need a rate here and do not need one for the traffic items above and below?

Comment (StepanVovkPLV) [reply DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_tagged_ports.py:101]: As multicast traffic goes to CPU, with many traffic items configured and `rate: 100` we will get losses

**Comment by StepanVovkPLV** [comment]
Closed in favor of https://github.com/PLVision/dent-testing/pull/42

**Pull request closed without merging**